pennylane/interfaces/execution.py

```diff
@@ -599,36 +693,15 @@ def gradient_fn(internal_tapes):
         # in this case would have ambiguous behaviour.
         raise ValueError("Gradient transforms cannot be used with grad_on_execution=True")
 
-    mapped_interface = INTERFACE_MAP[config.interface]
+    ml_boundary = _get_ml_framework_boundary(interface, _grad_on_execution)
     try:
-        if mapped_interface == "autograd":
-            from .autograd import execute as _execute
-
-        elif mapped_interface == "tf":
-            import tensorflow as tf
-
-            if not tf.executing_eagerly() or "autograph" in interface:
-                from .tensorflow_autograph import execute as _execute
-
-                _execute = partial(_execute, grad_on_execution=_grad_on_execution)
-
-            else:
-                from .tensorflow import execute as _execute
-
-        elif mapped_interface == "torch":
-            from .torch import execute as _execute
-
-        elif mapped_interface == "jax":
-            _execute = _get_jax_execute_fn(interface, tapes)
-
-        res = _execute(
+        res = ml_boundary(

```

Comment:
I don't really agree with naming it ml_boundary, I think you should add the purpose of the function in the name maybe `ml_boundary_execute` or `ml_execute`. If you are not familiar with the codebase you cannot directly understand what happen to the tapes.